From c86e19ef02a7faf845b399a50f4aa0b23b003fcf Mon Sep 17 00:00:00 2001
From: Syam Sidhardhan <s.syam@samsung.com>
Date: Mon, 16 Apr 2012 18:31:38 +0530
Subject: [PATCH 1/2] eir: Fix incorrect eir_has_data_type() parsing

Updating the "parsed" variable twice inside the for loop, leads to
incorrect parsing.
---
 src/eir.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/eir.c b/src/eir.c
index 419f444..4dfadea 100644
--- a/src/eir.c
+++ b/src/eir.c
@@ -337,9 +337,9 @@ void eir_create(const char *name, int8_t tx_power, uint16_t did_vendor,
 gboolean eir_has_data_type(uint8_t *data, size_t len, uint8_t type)
 {
 	uint8_t field_len;
-	size_t parsed;
+	size_t parsed = 0;
 
-	for (parsed = 0; parsed < len - 1; parsed += field_len) {
+	while (parsed < len - 1) {
 		field_len = data[0];
 
 		if (field_len == 0)
-- 
1.7.7.3

